### PR TITLE
New shinyapps deployments can be immediately private

### LINKED ISSUE #403: No option to set visibility (private/public) when deploying to shinyapps.io
I think it would be really useful to have an option in rsconnect::deployApp() that enabled setting of public/private visibility. At the moment publishing to shinyapps.io defaults to public and if you want it to be private you have to then remember to set it in the dashboard manually afterwards while your application is already open to the public. Found 2015 post on the subject by @tareefk [here](https://groups.google.com/forum/#!topic/shinyapps-users/--CE88p_IxU) (shinyapps.io google group).

### PR DESCRIPTION
**Pull request by mbaynton**
Fixes #403

### CODE REVIEW

File: R/deployApp.R [hunk 1]
```diff
@@ -351,6 +353,20 @@ deployApp <- function(appDir = getwd(),
     application <- applicationForTarget(client, accountDetails, target, forceUpdate)
   })
 
+  # Shinyapps defaults to public visibility.
+  # Other values should be set before data is deployed.
+  currentVisibility <- application$deployment$properties$application.visibility
+  if (identical(currentVisibility, NULL)) {
```
Comment (aronatkins) [inline R/deployApp.R:359]: I think you want `is.null(currentVisibility)`

File: R/deployApp.R [hunk 2]
```diff
@@ -351,6 +353,20 @@ deployApp <- function(appDir = getwd(),
     application <- applicationForTarget(client, accountDetails, target, forceUpdate)
   })
 
+  # Shinyapps defaults to public visibility.
+  # Other values should be set before data is deployed.
+  currentVisibility <- application$deployment$properties$application.visibility
```
Comment (aronatkins) [inline R/deployApp.R:358]: Move computation of `currentVisibility` into your `isShinyapps` block. Only shinyapps has that data structure.

**Review by aronatkins — COMMENTED** [review]
Please update NEWS.md as well.

File: R/deployApp.R [hunk 3]
```diff
@@ -114,7 +114,9 @@ deployApp <- function(appDir = getwd(),
                       forceUpdate = getOption("rsconnect.force.update.apps", FALSE),
                       python = NULL,
                       on.failure = NULL,
-                      forceGeneratePythonEnvironment = FALSE) {
+                      forceGeneratePythonEnvironment = FALSE,
+                      shinyAppsVisibility = "public"
```
Comment (kippandrew) [inline R/deployApp.R:118]: Seems a bit odd to include `shinyapps` in the parameter name. What about `appVisibility`? Obviously, this is a shinyapps.io specific flag, but we've never explicitly called out RSC specific flags.

File: R/deployApp.R [hunk 4]
```diff
@@ -68,6 +68,9 @@
 #' @param forceGeneratePythonEnvironment Optional. If an existing
 #'   `requirements.txt` file is found, it will be overwritten when this argument
 #'   is `TRUE`.
+#' @param appVisibility One of `"private"`, `"public"`, or `"hidden"`; indicates
```
Comment (kippandrew) [inline R/deployApp.R:71]: Let's omit the "hidden" parameter here, it's not relevant for shinyapps.io.

**Review by kippandrew — APPROVED** [review]

**Review by aronatkins — APPROVED** [review]

**Pull request merged**